mm/mm_heap/mm_realloc.c

```diff
@@ -213,6 +215,28 @@ FAR void *mm_realloc(FAR struct mm_heap_s *heap, FAR void *oldmem,
             }
         }
 
+      /* Note that a free node can't be smaller than SIZEOF_MM_FREENODE.
+       * We have to either take the whole node, or leave at least
+       * SIZEOF_MM_FREENODE bytes.
+       * When we can't leave SIZEOF_MM_FREENODE bytes, just take the whole
+       * node. It might end up with returning a larger chunk than the
+       * caller requested. It should be ok.
+       */
+      DEBUGASSERT(prevsize >= takeprev);
+      DEBUGASSERT(nextsize >= takenext);
+      if (prevsize - takeprev < SIZEOF_MM_FREENODE)
+        {
+          takeprev = prevsize;
+        }
+      if (nextsize - takenext < SIZEOF_MM_FREENODE)
```

Comment:
Coding standard requires that all terminating right braces be followed by a blank line.